src/main/java/com/example/demo/DBconfiguration/dbConfiguration.java

```diff
@@ -0,0 +1,65 @@
+package com.example.demo.DBconfiguration;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration

```

Comment:
we should prefer default configurations if available. so this class may not be needed as it is written.

we should define spring data source properties for respective profile and it will automatically be fetched.

you need to get an understanding of how spring is working and how can we make best use with minimal work

plz see depaolo/api project for this